### PR TITLE
[Endpoints] Add skeleton v2 sample

### PR DESCRIPTION
**Pull request by frankyn**
This is a PR to add a skeleton V2 sample.

### CODE REVIEW

File: appengine-java8/endpoints-v2-skeleton/pom.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+-->
+<project>
+    <modelVersion>4.0.0</modelVersion>
+    <packaging>war</packaging>
+    <version>1.0-SNAPSHOT</version>
+
+    <groupId>com.example.skeleton</groupId>
+    <artifactId>endpoints-j8-skeleton</artifactId>
+
+    <parent>
```
Comment (jabubake) [inline appengine-java8/endpoints-v2-skeleton/pom.xml]: If we are providing this as a skeleton app, maybe we should take out the parent reference here.

File: appengine-java8/endpoints-v2-skeleton/pom.xml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                            <filtering>true</filtering>
+                            <targetPath>WEB-INF</targetPath>
+                        </resources>
+                    </webResources>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>com.google.cloud.tools</groupId>
+                <artifactId>appengine-maven-plugin</artifactId>
+                <version>1.3.2</version>
```
Comment (jabubake) [inline appengine-java8/endpoints-v2-skeleton/pom.xml]: noticed you used 1.3.3 in the gradle version

File: appengine-java8/endpoints-v2-skeleton/pom.xml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                    <!-- deploy configuration -->
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>com.google.cloud.tools</groupId>
+                <artifactId>endpoints-framework-maven-plugin</artifactId>
+                <version>1.0.2</version>
+                <configuration>
+                    <!-- plugin configuration -->
+                    <hostname>YOUR_PROJECT_ID.appspot.com</hostname>
```
Comment (jabubake) [inline appengine-java8/endpoints-v2-skeleton/pom.xml]: should this be marked as "MUST REPLACE" ?

File: appengine-java8/endpoints-v2-skeleton/pom.xml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                </configuration>
+            </plugin>
+            <!--
+                 The versions-maven-plugin is used for dependency updates and
+                 is not required.
+            -->
+            <plugin>
```
Comment (jabubake) [inline appengine-java8/endpoints-v2-skeleton/pom.xml]: If this is not required and we cal this a skeleton app, maybe we can opt out of including it ? (making for a simpler `pom.xml`)

File: appengine-java8/endpoints-v2-skeleton/src/main/java/com/example/skeleton/MyApi.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package com.example.skeleton;
+
+import com.google.api.server.spi.config.Api;
+
+/**
+ * MyApi skeleton endpoints sample
+ * Add your first API methods in this class, or you may create another class.
+ * In that case, update your src/main/webapp/WEB-INF/web.xml accordingly.
```
Comment (jabubake) [inline appengine-java8/endpoints-v2-skeleton/src/main/java/com/example/skeleton/MyApi.java]: "In that case" comment, can  you rephrase that to, if you create another class ?

File: appengine-java8/endpoints-v2-skeleton/src/main/webapp/WEB-INF/appengine-web.xml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+<!-- [START appengine_web_xml] -->
+<appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
+    <runtime>java8</runtime>
+    <threadsafe>true</threadsafe>
+
+    <basic-scaling>
```
Comment (jabubake) [inline appengine-java8/endpoints-v2-skeleton/src/main/webapp/WEB-INF/appengine-web.xml]: Is this a requirement for the user in their skeleton app, if not, can we leave it out ?

**Comment by jabubake** [comment]
@frankyn : also, do you know why the tests failed ?

File: appengine-java8/endpoints-v2-skeleton/pom.xml (hunk 4 above)
Comment (frankyn) [reply appengine-java8/endpoints-v2-skeleton/pom.xml]: I guess I misunderstood, but I thought Dependency Auto-update used this plugin to know which dependencies to update?

File: appengine-java8/endpoints-v2-skeleton/src/main/webapp/WEB-INF/appengine-web.xml (hunk 6 above)
Comment (frankyn) [reply appengine-java8/endpoints-v2-skeleton/src/main/webapp/WEB-INF/appengine-web.xml]: Good call, I'll remove this.

File: appengine-java8/endpoints-v2-skeleton/pom.xml (hunk 1 above)
Comment (frankyn) [reply appengine-java8/endpoints-v2-skeleton/pom.xml]: I forgot to add this sample to the parent pom.xml.. The rationale for keeping it in is for dependency updates as IIRC it's dependent on the parent pom.

**Comment by frankyn** [comment]
Failure is from Firestore
```
[INFO] Google Cloud Firestore Samples ..................... FAILURE [ 11.197 s]
```

File: appengine-java8/endpoints-v2-skeleton/pom.xml (hunk 2 above)
Comment (frankyn) [reply appengine-java8/endpoints-v2-skeleton/pom.xml]: The maven appengine plugin is at 1.3.2

**Comment by frankyn** [comment]
I updated the PR to reflect your comments @jabubake PTAL. Thank you for your review!

File: appengine-java8/endpoints-v2-skeleton/pom.xml (hunk 1 above)
Comment (jabubake) [reply appengine-java8/endpoints-v2-skeleton/pom.xml]: looks like @kurtisvg is working on a related cleanup, so ignore my comment here.

File: appengine-java8/endpoints-v2-skeleton/pom.xml (hunk 2 above)
Comment (jabubake) [reply appengine-java8/endpoints-v2-skeleton/pom.xml]: 👍

File: appengine-java8/endpoints-v2-skeleton/pom.xml (hunk 4 above)
Comment (jabubake) [reply appengine-java8/endpoints-v2-skeleton/pom.xml]: we don't include it within each sample though:
already included here.
https://github.com/GoogleCloudPlatform/java-repo-tools/blob/master/pom.xml#L237

**Comment by jabubake** [comment]
LGTM, just a minor update required in removing the versions plugin from the pom.xml

File: appengine-java8/endpoints-v2-skeleton/pom.xml (hunk 4 above)
Comment (frankyn) [reply appengine-java8/endpoints-v2-skeleton/pom.xml]: ah thanks for clarifying! Will fix and then merge.

**Pull request merged**